File: Collatz/collatz_in_C/collatz.c
#include <stdio.h>
#include "collatz.h"

int main(void) {
  unsigned long n = N;
  printf(" n = %d\n", n);
  while ( n > 1 ) {
    if ( n % 2 == 0 ) {
      n = n / 2;
    }
    else {
      n = 3 * n + 1;
    }
    printf(" n = %ld\n", n);
    sleep(1);
  }
}

File: Collatz/collatz/collatz.f90
program collatz
  use constants_m
  implicit none

  integer(DI) :: n = 30

  print *, " n = ", n

  do while ( n > 1 ) 
    if ( mod( n, 2) == 0 ) then
      n = n / 2
    else
      n = 3 * n + 1
    end if
    print *, " n = ", n
    call sleep(1)
  end do

end program collatz
